### ISSUE TITLE
Allow Replay search to take an email address (without a key such as `user.email`)

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry#44957]
### Problem Statement

Users go into Sentry and paste an email. Currently, this runs a query and returns no results.

![image](https://user-images.githubusercontent.com/1633368/220686657-24361d9e-6941-4d70-a501-f8c62204dcf5.png)

After that, it shows the email in _Recent Searches_.

![image](https://user-images.githubusercontent.com/1633368/220686756-344ba0e9-1bb4-4895-9b49-18d679a0f288.png)

Users are surprised they can't find the events (this particularly user was referring to Replays but one could apply to errors, transactions).

### Solution Brainstorm

Similar to how we can paste a raw event id into a search in issues, we should allow an email.
Basically alias'ing to a search: `user.email:user@domain.tld`

**Comment by ryan953** [comment]
On issues i notice that if i put an issue id into the search box, like `JAVASCRIPT-24V3` it'll take me directly to that issue.

**Comment by bruno-garcia** [comment]
> On issues i notice that if i put an issue id into the search box, like `JAVASCRIPT-24V3` it'll take me directly to that issue.

Same with event ids. SDKs often print them, so one can easily paste into Sentry to go straight to the event

**Comment by ryan953** [comment]
One approach might be:

1. try to part the query, when the fails
2. regexp for emails -> if matches then rewrite the query to be user.email:${query} and go
3. isNumeric check -> if matches then rewrite for id:${query} and go
maybe there are other checks we can use to infer what they were looking for, and then fire off the usual query.

**Comment by c298lee** [comment]
Closing this issue because the cost is not worth the value we are adding. This would only help a small percentage of users, and it is more beneficial to teach our users how to use our search. This may also set the precedent that we will do this for other key:value pairs, which we won't. This change would also affect the grammar for the search query, which is legacy code. This would be a better solution: [https://github.com/getsentry/sentry/issues/53085](url)
